examples/tree_3d_dgsem/elixir_mhd_dynamo.jl

```diff
@@ -0,0 +1,165 @@
+
+    # Random phase -pi < phi <= pi
+    phi = (randn(rng) * 2 - 1) * pi
+
+    # Random vector k, also delta-correlated in time.
+    # k = SVector((randn(rng) * 2 - 1) * pi, (randn(rng) * 2 - 1) * pi, (randn(rng) * 2 - 1) * pi)
+    k = [0.0, 0.0, 0.0]
+    k[1] = (randn(rng) * 2 - 1) * pi
+    k[2] = (randn(rng) * 2 - 1) * pi
+    k[3] = (randn(rng) * 2 - 1) * pi
+    k = k / norm(k)
+    k = 6 * k
+    k_norm = k / norm(k)
+
+    # Random unit vector, not aligned with k.
+    ee = SVector(randn(rng), randn(rng), randn(rng)) .* 2 .- 1
+    # Normalize the vector e.
+    ee = ee / norm(ee)
+
+    # Compute the f-vectors.
+    f_nohel = cross(k, ee) / sqrt(dot(k, k) - dot(k, ee)^2)
+    M = [0 k_norm[3] -k_norm[2]; -k_norm[3] 0 k_norm[1]; k_norm[2] -k_norm[1] 0]
+    R = (I - im * sigma * M) / sqrt(1 + sigma^2)
+    f_k = R * f_nohel
+
+    # Normalization factor to make sure that the time integration of f is 0.
+    N = f_0 * c_s * sqrt(norm(k) * c_s / delta_t)
+
+    forcing = real(N * f_k * exp(im * dot(k, x) + im * phi)) / u[1]
+
+    return SVector(0, forcing[1], forcing[2], forcing[3], 0, 0, 0, 0, 0)
+end
+
+
+###############################################################################
+# semidiscretization of the resistive compressible MHD equations
+
+prandtl_number() = 0.72
+# Corresponds to previous alpha^2 runs.
+mu() = 2e-3
+eta = 2e-3
+
+gamma = 1.0 + 2.0 / 3.0
+
+equations = IdealGlmMhdEquations3D(gamma)
+equations_parabolic = CompressibleMhdDiffusion3D(equations, mu = mu(),
+                                                 Prandtl = prandtl_number(), eta = eta,
+                                                 gradient_variables = GradientVariablesPrimitive())
+
+volume_flux = (flux_hindenlang_gassner, flux_nonconservative_powell)
+solver = DGSEM(polydeg = 3,
+               surface_flux = (flux_lax_friedrichs, flux_nonconservative_powell),
+               volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
+
+coordinates_min = (-pi / 4, -pi / 4, 0.0) # minimum coordinates (min(x), min(y), min(z))
+coordinates_max = (pi / 4, pi / 4, pi / 2) # maximum coordinates (max(x), max(y), max(z))
+
+# Create a uniformly refined mesh with periodic boundaries
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 3,
+                periodicity = (true, true, true),
+                n_cells_max = 100_000) # set maximum capacity of tree data structure
+
+initial_condition = initial_condition_gaussian_noise_mhd
+source_terms = source_terms_helical_forcing
+
+semi = SemidiscretizationHyperbolicParabolic(mesh, (equations, equations_parabolic),
+                                             initial_condition, solver,
+                                             source_terms = source_terms)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+# Create ODE problem with time span `tspan`
+tspan = (0.0, 0.001)
```

Comment:
This final time also seems very short especially for a turbulent box. What is true desired final time? If this shortened time is just for the testing you can pass `tspan=(0.0, 0.001)` to the appropriate `@test_trixi_include` call.

Reply:
For a turbulent dynamo we would need a rather long time. Unrealistically long for tests. In the spirit of other tests, I keep this time short.

Reply:
I totally understand that. What is the target final time for the test case? You could add a comment here that says what you wrote above. One `tspan` is artificially short for testing /  debugging and then a commented `tspan` could be labeled as the "real" time span for the turbulent box.

Reply:
Why was this resolved? You did not answer my question. What is the intended final time of this test case? Can it be found in the paper you referenced above? If that is the case, point the user to this information. Right now, this elixir seems nonsensical to invest all this effort to only integrate until a final time of 0.001.

As I mentioned in a previous comment, you can set a shortened end time within the `trixi_test` call. There is not necessarily any need to set this shorter time within the actual elixir file.

Reply:
The time span has been changed to 10.0 in the elixir and to 0.001 in the test.